src/logic/__tests__/saveProjectTest.js

```diff
@@ -0,0 +1,44 @@
+import saveProject from '../saveProject';
+import reduce from '../../reducers';
+import {userAuthenticated} from '../../actions/user';
+import {projectCreated, updateProjectSource} from '../../actions/projects';
+
+import {
+  credentialFactory,
+  userFactory,
+} from '@factories/clients/firebase';
+
+const defaultState = reduce(undefined, {type: 'NONE'});
+
+jest.mock('../../clients/firebase.js');
+
+test('not pristine project should save', async() => {
+  const mockCredential = credentialFactory.build();
+  const mockUser = userFactory.build();
+  let state = reduce(defaultState, projectCreated('123456'));
+  state = reduce(state, updateProjectSource('123456', 'css', ''));
+  state = reduce(state, userAuthenticated(mockUser, [mockCredential]));
+  const getState = jest.fn(() => state);
+  const {type} = await saveProject.process({getState});
+  expect(getState).toHaveBeenCalledWith();
+  expect(type).toBe('PROJECT_SUCCESSFULLY_SAVED');
+});
+
+test('pristine project should not save', async() => {
+  const mockCredential = credentialFactory.build();
+  const mockUser = userFactory.build();
+  let state = reduce(defaultState, projectCreated('123456'));
+  state = reduce(state, userAuthenticated(mockUser, [mockCredential]));
+  const getState = jest.fn(() => state);
+  const result = await saveProject.process({getState});
+  expect(getState).toHaveBeenCalledWith();
```

Comment:
It doesn’t hurt anything really, but I generally don’t explicitly assert that side-effect-free mocks have been called—generally we can infer that they have been called based on the rest of the test passing (concretely: if the logic weren’t getting the information provided by `getState`, then `result` would not be the right thing).